### PR TITLE
Timesketch terraform

### PR DESCRIPTION
**Pull request by berggren**
This is the initial PR to add Terraform deployment support for Timesketch (https://github.com/google/timesketch. Timesketch is a digital forensics timeline analysis tool.

@blueandgold Lets chat about the folder structure, I'm happy to change.

This PR will be updated with:
- [x] Documentation.
- [x] Inline comments where it makes sense.
- [x] Folder structure is still to be decided.

### CODE REVIEW

**Comment by berggren** [comment]
@hshin-g @blueandgold Ready for a first round of reviews. FYI: I will be adding more documentation in followup PRs.

File: contrib/incident-response/infrastructure/timesketch/main.tf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  # Use Google Cloud Storage for robust, collaborative state storage.
+  # Note: The bucket name need to be globally unique.
+  #backend "gcs" {
+  #  bucket = "GLOBALLY UNIQ BUCKET NAME"
+  #}
+}
+
+provider "google" {
+  project  = "${var.gcp_project}"
```
Comment (hshin-g) [inline contrib/incident-response/infrastructure/timesketch/main.tf]: Should add param for credentials.

**Comment by hshin-g** [comment]
@berggren Got the following errors when trying to run `terraform plan`: 
```2 error(s) occurred:

* google_redis_instance.redis: 1 error(s) occurred:

* google_redis_instance.redis: Error creating Instance: googleapi: Error 403: Permission 'redis.instances.create' denied on 'projects/hshin-timesketch-1/locations/us-central1/instances/redis'
* google_compute_address.timesketch-server-address: 1 error(s) occurred:

* google_compute_address.timesketch-server-address: Error creating Address: googleapi: Error 403: Required 'compute.addresses.create' permission for 'projects/hshin-timesketch-1/regions/us-central1/addresses/timesketch-server-address', forbidden
```

File: contrib/incident-response/infrastructure/timesketch/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+```bash
+terraform init
+```
+
+Then run (`timesketch-deployment-test` is the name of a project that you've previously
+set up):
+
+```bash
+terraform apply -var 'gce_project=timesketch-deployment-test'
```
Comment (hshin-g) [inline contrib/incident-response/infrastructure/timesketch/README.md]: What is the `gce_project` variable used for? When you run `terraform plan/apply`, it asks you for `gcp_project`value. Are those interchangeable?

File: contrib/incident-response/infrastructure/timesketch/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+Run the following to get the URL of a newly deployed Timesketch server:
```
Comment (hshin-g) [inline contrib/incident-response/infrastructure/timesketch/README.md]: Navigated to the URL that was outputted, and have "took too long to respond" error message. Are there other instructions that should be included to get the URL working?

Comment (berggren) [reply contrib/incident-response/infrastructure/timesketch/README.md]: this have been fixed in the latest commit. It was a config param for the SQL server that prevented the server to start up.

File: contrib/incident-response/infrastructure/timesketch/README.md (hunk 2 above)
Comment (berggren) [reply contrib/incident-response/infrastructure/timesketch/README.md]: This is the name of the project you want to deploy to. I made this clearer in the description.

File: contrib/incident-response/infrastructure/timesketch/main.tf (hunk 1 above)
Comment (berggren) [reply contrib/incident-response/infrastructure/timesketch/main.tf]: Updated README.md addresses this.

File: contrib/incident-response/infrastructure/main.tf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  # Use local state storafe by default. For production environments please
```
Comment (hshin-g) [inline contrib/incident-response/infrastructure/main.tf]: storafe --> storage

Comment (berggren) [reply contrib/incident-response/infrastructure/main.tf]: Fixed

**Comment by berggren** [comment]
I'm OK with my commits being contributed to this project

**Pull request merged**